### DISCUSSION TITLE
Expose the probabilities of the next candidate tokens (logits/log-probs) via the high level API?

### DISCUSSION THREAD

**Discussion by stduhpf** [abetlen/llama-cpp-python#1101] [Ideas]
I think it would be useful to have a way to easily acess the next candidate tokens, for example to compare directly the probabilities of "yes" and "no" when doing sentiment anylysis. Right now I'm using  `model.sample(temp=0)`, and hope that the output is either of the two expected values, but sometimes the models decide they wants to put something else here, even at low temperature and with multi-shot.
I'm sure it must be possible with the low-level API, but it would be nice to have a convenient way to achieve this.

**Comment by ekorman** [comment]
i believe these are in the arrays `llm.scores` / `llm._scores` after running inference
